### ISSUE TITLE
Running the aws-service-operator outside of AWS

### ISSUE THREAD

**Issue by Samze** [awslabs/aws-service-operator#122]
Hey,

We noticed in the documentation it states that to deploy the aws service operator you must have `a Kubernetes cluster running in AWS`.

Is there a particular reason for this and are there plans to support non-aws k8s clusters?

Cheers,
Sam & @teddyking

**Comment by tantona** [comment]
It's _possible_ to run the aws-service-operator in non-aws clusters. I develop on my machine with `docker-for-desktop` (macOS).  What is your use case?

**Comment by davidxjohnson** [comment]
My opinion as an interested party (not speaking for the contributors). 

One of the value-add (for me) is to eliminate the disjointed deploy/destroy of K8S applications running on AWS. In other words, get rid of the two-step process having to run CF templates (or Ansible  or Terraform) to create the resources outside of K8S before installing the app (the same goes for tear-down). This problem is solved by representing AWS resources as CRDs in K8S, thus making it possible to use the same tooling for both K8S and AWS - i.e. kubectl, Helm, Helmfile etc. 

Representing the AWS object as CRD doesn't make a lot of sense (to me) outside the context of AWS. Let's take the example of a service in K8S that deploys/destroys an ELB/NLB/ALB on AWS. Why doesn't this work on K8S clusters not running on AWS?

**Comment by christopherhein** [comment]
@Samze any interest in modifying that text to reflect that this could be run anywhere?

The main thing that might change about this later is when we have in VPC resources, for example, an RDS that requires subnets in your VPC to be provisioned.

Not to say we couldn't use some mechanisms around this for non-AWS running clusters just more out of scope for the current configurations.

**Comment by pawelprazak** [comment]
A use case I'm interested in is bootstrapping an EKS cluster. 

So:
- a person runs the `aws-service-operator` with a CR describing an EKS cluster
- an EKS cluster gets created
- `aws-service-operator` installs itself on the newly created EKS cluster

Basically a self-deployment mechanism.

**Comment by ckrough** [comment]
@pawelprazak There is https://github.com/awslabs/aws-eks-cluster-controller, but the parent-child relationship there depends on AWS IAM for auth, etc., so it is dependent on the clusters all being in AWS.

**Comment by pawelprazak** [comment]
yes, but the first cluster is created manually mostly, you are just moving the problem elsewhere, with self-bootstrap it would be all automatic
